LiteCore/Database/CollectionImpl.hh

```diff
@@ -210,7 +210,11 @@ namespace litecore {
                 } while ( doc->selectNextRevision() );
                 if ( !revID ) return false;
             }
-            doc->setRemoteAncestorRevID(remoteID, revID);
+            if ( remoteID == RevTree::kNoRemoteID ) {
+                doc->revIsRejected(revID);
```

Comment:
Why need to call this function here?

Reply:
c.f. Pusher+Revs.cc, where we received 403 from SG. There, we don't have Rev object itself, we only work with ReplicatedRev data structure where Rev is represented as RevID. This is the place where we do book-keeping as the rev is finally synced. In this case, it is kind-of "synced" in the sense that it is formally rejected by SG.
That is, as a rev comes to here, either, it is normally synced, and we mark the remote ancestor. Or, it is abnormally synced, or formally rejected, we remember this fact.

Reply:
It's just a weird to call the getter method to do something here.

Reply:
Wait, it's not the getter. Sorry, I understood this wrong.